src/connectors/nginx.py

```diff
@@ -0,0 +1,290 @@
+        'prefix': "s3://",
+        'placeholder': "my-nginx-s3-bucket",
+    },
+    {
+        'type': 'str',
+        'name': 'aws_role',
+        'title': "Bucket Reader Role",
+        'prompt': "Role to be assumed for access to nginx log files in S3",
+        'placeholder': "arn:aws:iam::012345678987:role/my-nginx-read-role",
+    },
+    {
+        'type': 'str',
+        'name': 'prefix',
+        'title': "Log Prefix Filter",
+        'prompt': "Folder in S3 bucket where nginx access and error folders are located",
+        'default': "operational_logs/",
+    },
+    {
+        'type': 'str',
+        'name': 'existing_stage',
+        'title': "Snowflake Stage (alternative)",
+        'prompt': "Enter to use an existing stage instead",
+        'placeholder': "snowalert.data.nginx_stage",
+    },
+]
+
+
+LOG_LANDING_TABLE_COLUMNS = [
+    ("raw", "VARIANT"),
+    ("hash_raw", "NUMBER(19,0)"),
+    ("request_id", "VARCHAR(16777216)"),
+    ("event_time", "TIMESTAMP_LTZ(9)"),
+    ("bytes_sent", "NUMBER(38,0)"),
+    ("connection", "NUMBER(38,0)"),
+    ("connection_requests", "NUMBER(38,0)"),
+    ("deployment_cluster", "VARCHAR(16777216)"),
+    ("gzip_ratio", "FLOAT"),
+    ("host_header", "VARCHAR(16777216)"),
+    ("host_name", "VARCHAR(16777216)"),
+    ("user_agent", "VARCHAR(16777216)"),
+    ("http_user_agent", "VARCHAR(16777216)"),
+    ("http_xff", "VARCHAR(16777216)"),
+    ("http_referer", "VARCHAR(16777216)"),
+    ("http_method", "VARCHAR(16777216)"),
+    ("instance_id", "VARCHAR(16777216)"),
+    ("redirect_counter", "NUMBER(38,0)"),
+    ("remote_address", "VARCHAR(16777216)"),
+    ("request", "VARCHAR(16777216)"),
+    ("request_time", "FLOAT"),
+    ("requests_length", "NUMBER(38,0)"),
+    ("ssl_session_id", "VARCHAR(16777216)"),
+    ("ssl_session_reused", "VARCHAR(16777216)"),
+    ("status", "NUMBER(38,0)"),
+    ("event_time_other", "VARIANT"),
+    ("n_upstream_attempts", "NUMBER(38,0)"),
+    ("upstream_address", "VARCHAR(16777216)"),
+    ("upstream_response_length", "NUMBER(38,0)"),
+    ("upstream_response_time", "FLOAT"),
+    ("upstream_status", "NUMBER(38,0)"),
+]
+
+ERROR_LANDING_TABLE_COLUMNS = [
+    ("raw", "VARIANT"),
+    ("hash_raw", "NUMBER(19,0)"),
+    ("instance_id", "VARCHAR(16777216)"),
+    ("log_level", "VARCHAR(16777216)"),
+    ("message", "VARCHAR(16777216)"),
+    ("event_time", "TIMESTAMP_LTZ(9)"),
+    ("pid", "NUMBER(38,0)"),
+    ("tid", "NUMBER(38,0)"),
+]
+
+CONNECT_RESPONSE_MESSAGE = """
+STEP 1: Be sure the Role "{role}" includes the following trust relationship:
+
+{role_trust_relationship}
+
+
+STEP 2: For Role "{role}", ensure the following inline policy:
+
+{role_policy}
+"""
+
+
+def connect(connection_name, options):
+    table_name = f'nginx_{connection_name}_connection'
+    log_landing_table = f'data.{table_name}'
+    error_landing_table = f'data.nginx_{connection_name}_error_connection'
+    prefix = ''
+    bucket_name = ''
+
+    for i in options:
```

Comment:
I think options is always a dict so this is more clear as

    if '' in options:
        del options['']

Reply:
That doesn't have the same effect.

`>>> c = {'foo': '', 'bar': 'baz'}
>>> if '' in c:
... 	del c['']
...
>>> c
{'foo': '', 'bar': 'baz'}`

Reply:
`for i in option` iterates over the keys, not the values. if you're looking to remove empty values, it's probably clearest to re-construct the dict instead of mutating it, i.e.

    d = {k: v for k, v in d.items() if v != ''}